### PR TITLE
My Jetpack: Add scripts for JS tests

### PR DESCRIPTION
**Pull request by renatoagds**
Include scripts for `test`, `watch` and `coverage` related with JS files.

#### Changes proposed in this Pull Request:

* Add `test` script into `package.json`.
* Add `test-js`, `test-js-watch` into `composer.json`.
* Add JS coverage into `test-coverage`.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

_To run general all tests_

* Run `jetpack test packages/my-jetpack js`
* Tests will run

_To enter watch mode_

* Run `jetpack test packages/my-jetpack js-watch`
* It will enter into watch mode.

_To run coverage_

* Run `jetpack test packages/my-jetpack coverage`
* Tests will run and `coverage` folder will be created.

_To run with COVERAGE_DIR_

* Run `COVERAGE_DIR=agora jetpack test packages/my-jetpack coverage`
* Tests will run and `agora` folder will be created.

### CODE REVIEW

File: projects/packages/my-jetpack/composer.json [hunk 1]
```diff
@@ -23,12 +23,24 @@
 		],
 		"test-coverage": [
 			"@composer install",
-			"php -dpcov.directory=. ./vendor/bin/phpunit --coverage-clover \"$COVERAGE_DIR/clover.xml\""
+			"php -dpcov.directory=. ./vendor/bin/phpunit --coverage-clover \"$COVERAGE_DIR/clover.xml\"",
+			"pnpm install",
+			"pnpm run test -- --coverageDirectory=\"$COVERAGE_DIR\" --coverage"
```
Comment (anomiex) [inline projects/packages/my-jetpack/composer.json:28]: As written both of these will write to `$COVERAGE_DIR/clover.xml`, so the JS coverage will overwrite the PHP coverage. You should arrange for them to have different names.

Also, you may want to pass `--coverageReporters=clover` for the JS so it won't also generate an lcov report.

Comment (renatoagds) [reply projects/packages/my-jetpack/composer.json:28]: @anomiex Yes! I was seeing now the crash between PHP and JS, pushing a fix for that.

Comment (renatoagds) [reply projects/packages/my-jetpack/composer.json]: done.

**Comment by oskosk** [comment]
I'm not sure we should be bumping version for the package when adding a change in the scripts for running tests. 

Looks to me like `projects/packages/my-jetpack/changelog/update-my-jetpack-unit-tests-scripts` should be `patch`.

**Review by retrofox — APPROVED** [review]
LGTM.

**Comment by retrofox** [comment]
I didn't see Osk's comment [here](https://github.com/Automattic/jetpack/pull/22358#issuecomment-1013666026). We should consider taking a look before merging it?

**Review by retrofox — DISMISSED** [review]
Removing ✅ because of [this comment](https://github.com/Automattic/jetpack/pull/22358#issuecomment-1013666026). It is worths taking a look at it.

**Review by retrofox — APPROVED** [review]
Tested, and LGTM. I suggest another ✅ from folks, but I think it's ready to take off. We can continue a a follow-up

```cli
 🦊 retrofox  ~/lab/jetpack   update/my-jetpack-unit-tests-scripts  jetpack test packages/my-jetpack js
Running test-js tests for packages/my-jetpack
> Composer\Config::disableProcessTimeout
> pnpm install
Scope: all 34 workspace projects
../../.. prepare$ husky install
│ husky - Git hooks installed
└─ Done in 1.1s
> pnpm run test
```

**Comment by renatoagds** [comment]
@retrofox just needed to update the `composer.lock` :)

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by retrofox** [comment]
> Great news! One last step: head over to your WordPress.com diff, D73110-code, and commit it. Once you've done so, come back to this PR and add a comment with your changeset ID.

r238336-wpcom

**Comment by oskosk** [comment]
Thanks for removing the version bump @retrofox @renatoagds

**Pull request merged**